assets/js/modules/analytics-4/components/settings/SettingsEnhancedMeasurementSwitch.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * SettingsEnhancedMeasurementSwitch component.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
+import {
+	ENHANCED_MEASUREMENT_ENABLED,
+	ENHANCED_MEASUREMENT_FORM,
+	MODULES_ANALYTICS_4,
+	PROPERTY_CREATE,
+	WEBDATASTREAM_CREATE,
+} from '../../datastore/constants';
+import EnhancedMeasurementSwitch from '../common/EnhancedMeasurementSwitch';
+import {
+	isValidPropertyID,
+	isValidPropertySelection,
+	isValidWebDataStreamID,
+	isValidWebDataStreamSelection,
+} from '../../utils/validation';
+const { useSelect, useDispatch } = Data;
+
+export default function SettingsEnhancedMeasurementSwitch( {
+	hasAnalytics4Access,
+} ) {
+	const isEnhancedMeasurementEnabled = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue(
+			ENHANCED_MEASUREMENT_FORM,
+			ENHANCED_MEASUREMENT_ENABLED
+		)
+	);
+
+	const propertyID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getPropertyID()
+	);
+
+	const webDataStreamID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getWebDataStreamID()
+	);
+
+	const isLoadingProperties = useSelect( ( select ) => {
+		return select( MODULES_ANALYTICS_4 ).isLoadingProperties( {
+			hasModuleAccess: hasAnalytics4Access,
+		} );
+	} );
+
+	const isLoadingWebDataStreams = useSelect( ( select ) => {
+		return select( MODULES_ANALYTICS_4 ).isLoadingWebDataStreams( {
+			hasModuleAccess: hasAnalytics4Access,
+		} );
+	} );
+
+	const isEnhancedMeasurementStreamEnabled = useSelect( ( select ) => {
+		if (
+			! isValidPropertyID( propertyID ) ||
+			! isValidWebDataStreamID( webDataStreamID ) ||
+			isLoadingProperties ||
```

Comment:
As mentioned above, we need to wait for the `PropertySelect` and `WebDataStreamSelect` loading states to settle, in part to synchronise the loading from a UI perspective, but also so we can avoid inadvertently making a request for the enhanced measurement settings for the wrong combination of `propertyID` and `webDataStreamID`.